src/clang_options_parser/clang_options_parser.py

```diff
@@ -0,0 +1,228 @@
+# *******************************************************************************
+#                                                                              *
+#  This file is part of Verificarlo.                                           *
+#                                                                              *
+#  Copyright (c) 2015-2021                                                     *
+#     Verificarlo contributors                                                 *
+#     Universite de Versailles St-Quentin-en-Yvelines                          *
+#     CMLA, Ecole Normale Superieure de Cachan                                 *
+#                                                                              *
+#  Verificarlo is free software: you can redistribute it and/or modify         *
+#  it under the terms of the GNU General Public License as published by        *
+#  the Free Software Foundation, either version 3 of the License, or           *
+#  (at your option) any later version.                                         *
+#                                                                              *
+#  Verificarlo is distributed in the hope that it will be useful,              *
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of              *
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               *
+#  GNU General Public License for more details.                                *
+#                                                                              *
+#  You should have received a copy of the GNU General Public License           *
+#  along with Verificarlo.  If not, see <http://www.gnu.org/licenses/>.        *
+#                                                                              *
+# *******************************************************************************
+
+
+target_specific_options_x86 = set([
```

Comment:
is this list taken from somewhere? if yes that would be nice at least in comment to cite the source. And check the license if relevant.

Reply:
It was taken from `gcc --help-target`. Normally, gcc flags are compatible with clang but we could miss some. I would like to have an automate process for handling those options.

Reply:
Can we get the same list from the LLVM project, instead of taking a GPL header file ? I see no other way around.

Reply:
Ok I found an other way to solve the issue without parsing the inputs flags so we can remove this file. I'm just not sure if it preserves the optimization flags so we should discuss about it. The idea is to check if caller (source code) and callee (vfcwrapper) functions have the same ABI, meaning if they have the same `target-cpu` and `target-features` attributes. If not, I update the caller with the atttributes' callee. In that case, I always compile vfcwrapper with `-03 -march=native` to trigger vectorization. My main concern is if promoting the caller function may trigger optimization that are not specified by the user. Let's say it passes `-O0` but we optimize the function anyway. However, this case should not met in practice because passing -O0 as input flags should not trigger vectorization. I'll push my update soon.

Reply:
To precise why the previous solution was not working, within fuzzy the optimization flags used for compiling the Python Interpreter and those for the numpy packages are different. So the definition of the vectorized functions (`.vfcwrapper.o`) are different and the definition of the interpreter are taken instead of those of numpy. Causing to pass values in the wrong registers (xmm instead of zmm for avx512f). By always using the same optimization flags to compile the `vfcwrapper.c` we ensure that definitions are consistent across different library compilation.